Mathlib/NumberTheory/LSeries/Basic.lean

```diff
@@ -160,73 +248,104 @@ lemma LSeriesSummable_of_le_const_mul_rpow {f : ArithmeticFunction ℂ} {x : ℝ
-
-theorem LSeriesSummable_of_bounded_of_one_lt_real {f : ArithmeticFunction ℂ} {m : ℝ}
-    (h : ∀ n : ℕ, Complex.abs (f n) ≤ m) {z : ℝ} (hz : 1 < z) : f.LSeriesSummable z :=
-  LSeriesSummable_of_bounded_of_one_lt_re h <| by simp only [ofReal_re, hz]
-#align nat.arithmetic_function.l_series_summable_of_bounded_of_one_lt_real ArithmeticFunction.LSeriesSummable_of_bounded_of_one_lt_real
-
-open scoped ArithmeticFunction
-
-theorem zeta_LSeriesSummable_iff_one_lt_re {z : ℂ} : LSeriesSummable ζ z ↔ 1 < z.re := by
-  rw [← LSeriesSummable_iff_of_re_eq_re (Complex.ofReal_re z.re), LSeriesSummable, ←
-    summable_norm_iff, ← Real.summable_one_div_nat_rpow, iff_iff_eq]
-  by_cases h0 : z.re = 0
-  · rw [h0, ← summable_nat_add_iff 1]
-    apply congr rfl
-    ext n
-    simp [n.succ_ne_zero]
-  · apply congr rfl
-    ext ⟨- | n⟩
-    · simp [h0]
-    simp only [cast_zero, natCoe_apply, zeta_apply, succ_ne_zero, if_false, cast_succ, one_div,
-      Complex.norm_eq_abs, map_inv₀, Complex.abs_cpow_real, inv_inj, zero_add]
-    rw [← cast_one, ← cast_add, Complex.abs_natCast, cast_add, cast_one]
-#align nat.arithmetic_function.zeta_l_series_summable_iff_one_lt_re ArithmeticFunction.zeta_LSeriesSummable_iff_one_lt_re
+/-- If `f` is bounded, then its `LSeries` is summable at `s` when `re s > 1`. -/
+theorem LSeriesSummable_of_bounded_of_one_lt_re {f : ℕ → ℂ} {m : ℝ}
+    (h : ∀ n ≠ 0, Complex.abs (f n) ≤ m) {s : ℂ} (hs : 1 < s.re) :
+    LSeriesSummable f s := by
+  refine LSeriesSummable_of_le_const_mul_rpow hs ⟨m, fun n hn ↦ ?_⟩
+  simp only [norm_eq_abs, sub_self, Real.rpow_zero, mul_one, h n hn]
+#align nat.arithmetic_function.l_series_summable_of_bounded_of_one_lt_re LSeriesSummable_of_bounded_of_one_lt_re
+
+/-- If `f` is bounded, then its `LSeries` is summable at `s : ℝ` when `s > 1`. -/
+theorem LSeriesSummable_of_bounded_of_one_lt_real {f : ℕ → ℂ} {m : ℝ}
+    (h : ∀ n ≠ 0, Complex.abs (f n) ≤ m) {s : ℝ} (hs : 1 < s) :
+    LSeriesSummable f s :=
+  LSeriesSummable_of_bounded_of_one_lt_re h <| by simp only [ofReal_re, hs]
+#align nat.arithmetic_function.l_series_summable_of_bounded_of_one_lt_real LSeriesSummable_of_bounded_of_one_lt_real
+
+-- TODO: Move this to a separate file on concrete L-series
+
+open Set in
+/-- The `LSeries` with all coefficients `1` converges at `s`if and only if `re s > 1`. -/
+theorem LSeriesSummable.one_iff_one_lt_re {s : ℂ} : LSeriesSummable 1 s ↔ 1 < s.re := by
+  rw [← LSeriesSummable_iff_of_re_eq_re (Complex.ofReal_re s.re), LSeriesSummable,
+    ← summable_norm_iff, ← Real.summable_one_div_nat_rpow]
+  simp_rw [← Finite.summable_compl_iff (finite_singleton 0), summable_subtype_iff_indicator]
+  refine summable_congr fun n ↦ ?_
+  by_cases hn : n ∈ ({0}ᶜ :Set ℕ)
+  · have hn₀ : n ≠ 0 := hn
+    simp only [indicator_of_mem hn, norm_term_eq]
+    simp only [hn₀, ↓reduceIte, Pi.one_apply, norm_one, ofReal_re]
```

Comment:
```suggestion
  · simp only [indicator_of_mem hn, norm_term_eq]
    simp only [show n ≠ 0 from hn, ↓reduceIte, Pi.one_apply, norm_one, ofReal_re]
```
Just demonstrating a golfy hack in case you think it's worth it!